File: 数据结构与算法/快速排序/快速排序.js
function swap(arr, i, j) {
   var tmp = arr[i];
   arr[i] = arr[j];
   arr[j] = tmp;
}

function quickSort(a, low, high){
	var left, right, pv;
	if (low >= high) return;
	
	pv = a[low];
	left = low;
	right = high;
	
	while(left != right) {
		while(a[right] >= pv && right > left)
			right--;
		while(a[left] <= pv && right > left)
			left++;
			
		if (right > left) {
			swap(a, left, right)
		}
	}
	
	// 基数的交换
	swap(a, low, left);
	
	quickSort(a, low, left-1);
	quickSort(a, left+1, high); 
}

arr = [1, 8, 5, 33, 2, 3, 5, 7]
quickSort(arr, 0, arr.length-1)
console.log(arr)
    

File: 数据结构与算法/快速排序/快速排序.py
# 快速排序
def swap(L, a, b):
    tmp = L[a]
    L[a]= L[b]
    L[b] = tmp

def quick_sort(L, low, high):
    
    if low >= high:
        return
    
    tmp = L[low];
    left = low
    right = high
    
    while left < right:
        while right > left and L[right] >= tmp:
            right -= 1
        while left < right and L[left] <= tmp:
            left += 1
        
        if left != right:
            swap(L, left, right)
         
    if low != left:
        swap(L, left, low)
            
    quick_sort(L, low, left-1)
    quick_sort(L, left+1, high)
    

if __name__ == "__main__":
    # 读取n个数
    n = int(input())
    # 输入列表
    L = list(map(int, input().split()))
    
    print("排序前:")
    print(L)
    print("\n")
    quick_sort(L, 0, n-1)
    print("排序后:")
    print(L)
    print("\n")
